templates/register.html

```diff
@@ -2,6 +2,28 @@
 
 {% block content %}
 
+{% block javascript %}
+  <script>
+    $("#id_username").change(function () {
+      var username = $(this).val();
+
+      $.ajax({
+        url: '/ajax/validate_username/',
+        data: {
+          'username': username
+        },
+        dataType: 'json',
+        success: function (data) {
+          if (data.is_taken) {
+            alert("A user with this username already exists.");
+          }
+        }
+      });
+     $("#id_username").val('');
```

Comment:
no need to reset form field value. It's upto the user to change the form inputs value.